yt/utilities/io_handler.py

```diff
@@ -224,31 +224,34 @@ def _read_particle_selection(
 
     def _read_particle_fields(self, chunks, ptf, selector):
         # Now we have all the sizes, and we can allocate
-        data_files = set()
-        for chunk in chunks:
-            for obj in chunk.objs:
-                data_files.update(obj.data_files)
-        for data_file in sorted(data_files, key=lambda x: (x.filename, x.start)):
+        for data_file in self._sorted_chunk_iterator(chunks):
             data_file_data = self._read_particle_data_file(data_file, ptf, selector)
             # temporary trickery so it's still an iterator, need to adjust
             # the io_handler.BaseIOHandler.read_particle_selection() method
             # to not use an iterator.
             yield from data_file_data.items()
 
-
-# As a note: we don't *actually* want this to be how it is forever.  There's no
-# reason we need to have the fluid and particle IO handlers separated.  But,
-# for keeping track of which frontend is which, this is a useful abstraction.
-class BaseParticleIOHandler(BaseIOHandler):
-    def _sorted_chunk_iterator(self, chunks):
+    @staticmethod
+    def _get_data_files(chunks):
         chunks = list(chunks)
         data_files = set()
         for chunk in chunks:
             for obj in chunk.objs:
                 data_files.update(obj.data_files)
+        return data_files
+
+    def _sorted_chunk_iterator(self, chunks):
+        data_files = self._get_data_files(chunks)
```

Comment:
I think we should explicitly turn chunks into a list here, unless it has been elsewhere higher up the stack; recently we exhausted an iterator and ran into issues.  (Maybe with `nc_cm1`?)  Also, we sort with `x.start` and `x.filename` but not all frontends did that -- does `x.start` have a default value of 0 or something constant that we can rely on?

Reply:
Oh and I left the front ends that do not sort by both filename and start intact. They override this method so that they are unchanged

Reply:
OK, so `ParticleFile.start` does initialize to 0 if it's not provided (within `ParticleFile.__init__` ([link to relevant code](https://github.com/yt-project/yt/blob/8a113e9337ab526e3c7691ee6b2abd3d543696fc/yt/data_objects/static_output.py#L2074C8-L2087)). 

 But in any case, the changes in this PR do not change any behavior -- only the frontends that were sorting by `filename` and `start` are using this function. Those that were not already doing that either override this method or don't use this method at all.